src/pages/propose.astro

```diff
@@ -1,10 +1,151 @@
+        throw new Error('Name must be at least 6 characters long');
+      }
+
+      // Check if email is valid (you can add more complex validation)
+      if (email && !isValidEmail(email)) {
+        throw new Error('Invalid email address');
+      }
+
+      // Check if topic and summary have at least 6 characters
+      if ((topic && topic.length < 6) || (summary && summary.length < 6)) {
+        throw new Error('Topic and summary must be at least 6 characters long');
+      }
+
+      // Retrieve all selected values for checkboxes
+      const proposalTypes = data.getAll('proposal-type');
+
+      // This is POST request. It's not going to be visible in browser.
+      // Check Network tab in Inspect Mode of Google Chrome or Firefox to see request.
+    } catch (error) {
+      if (error instanceof Error) {
+        console.error(error.message);
+      }
+    }
+  }
+}
+
+// Add a simple email validation function
+function isValidEmail(email: string | null): boolean {
+  // Just check if it contains '@'
+  return !!email && email.includes('@');
+}
 ---
 
-<Layout title="Tulsa Web Devs">
-  <main class="grid w-screen h-screen bg-white text-4xl place-items-center">
-    Proposal Features Here!
+<Layout title="propose">
+  <main>
+    <img
+      alt="Tulsa Web Devs logo"
+      src="/twd-logo.svg"
+      height="auto"
+      width="100%"
+    />
+
+    <div class="flex flex-col items-center m-2" id="header">
+      <h1 class="text-3xl max-md:text-2xl">Propose an idea!</h1>
+      <h2 class="text-2xl max-md:text-xl">
+        Have an idea?! Submit it! Just fill out this form!
+      </h2>
+    </div>
+
+    <form method="POST" id="proposalForm" class="m-4">
+      <div class="p-4 flex flex-col">
+        <div id="name-fields" class="shrink p-2">
+          <label for="name" class="text-l font-semibold">Name:</label>
+          <input
+            type="text"
+            name="name"
+            required
+            minlength={6}
+            placeholder="Enter your name, nickname, or alias..."
+            class="rounded-small w-3/4 max-md:w-full"
+          />
+        </div>
+        <div id="email-field" class="shrink p-2">
+          <label for="email" class="text-l font-semibold">Email:</label>
+          <input
+            type="email"
+            name="email"
+            placeholder="myemailaddress@example.com..."
+            class="rounded-small w-3/4 max-md:w-full"
+          />
+        </div>
+      </div>
+
+      <div class="p-4" id="topic-summary-fields">
+        <label for="topic" class="text-l font-semibold">Topic:</label>
+        <textarea
+          id="topic"
+          name="topic"
+          required
+          minlength={6}
+          class="rounded-sm w-full"
+          placeholder="Topic title, framework name, main subject..."></textarea>
+        <label for="summary" class="text-l font-semibold">Summary:</label>
+        <textarea
+          id="summary"
+          name="summary"
```

Comment:
question: The [api spec](tulsawebdevs.org/docs/api) specifies both a 'summary' field and a 'description' field. Which is this meant to be?
suggestion: If it's a summary, we might add a max-length of ~200 characters, give or take, update the placeholder accordingly, and add a field for 'description' with no limit (or a higher one).
suggestion: If it's a description, we should rename the field to `description` and add a `summary` field (see above suggestion)

Reply:
I love this idea!

I was able to make the maxlength prop on the summary 200 characters.

I created a description field, with a maxlength prop of 600 characters. 800 chars total!

![image](https://github.com/tulsawebdevs/website/assets/98660173/f0e97b4e-5f61-4c41-b6ed-c5fe1e8a3b8b)